algebird-core/src/main/scala/com/twitter/algebird/SpaceSaver.scala

```diff
@@ -18,6 +21,98 @@ object SpaceSaver {
   private[algebird] val ordering = Ordering.by[(_, (Long, Long)), (Long, Long)]{ case (item, (count, err)) => (-count, err) }
 
   implicit def spaceSaverSemiGroup[T]: Semigroup[SpaceSaver[T]] = new SpaceSaverSemigroup[T]
+
+  /**
+   * Encodes the SpaceSaver as a sequence of bytes containing in order
+   * - 1 byte: 1/2 => 1 = SSOne, 2 = SSMany
+   * - 4 bytes: the capacity
+   * - N bytes: the item/counters (counters as length + N*(item size + item + 2 * counters)
+   */
+  def toBytes[T](ss: SpaceSaver[T], tSerializer: T => Array[Byte]): Array[Byte] = {
```

Comment:
Another ask - could we serialize a couple of examples and put a base64-encoded string of the bytes into the test, so that we can guarantee that the serialization doesn't change? I can see us making a change here that wouldn't break the tests, but would break deserialization of old instances.